mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,617 @@
+
+        if value_type is None:
+            raise Exception(f"Value type {type(value)} is not supported by weaviate!")
+
+        return value_type
+
+    def _translate_condition(
+        self,
+        table_name: str,
+        conditions: List[FilterCondition] = None,
+        meta_conditions: List[FilterCondition] = None,
+    ) -> Optional[dict]:
+        """
+        Translate a list of FilterCondition objects a dict that can be used by Weaviate.
+        E.g.,
+        [
+            FilterCondition(
+                column="metadata.created_at",
+                op=FilterOperator.LESS_THAN,
+                value="2020-01-01",
+            ),
+            FilterCondition(
+                column="metadata.created_at",
+                op=FilterOperator.GREATER_THAN,
+                value="2019-01-01",
+            )
+        ]
+        -->
+        {"operator": "And",
+        "operands": [
+            {
+                "path": ["created_at"],
+                "operator": "LessThan",
+                "valueText": "2020-01-01",
+            },
+            {
+                "path": ["created_at"],
+                "operator": "GreaterThan",
+                "valueInt": "2019-01-01",
+            },
+        ]}
+        """
+        table_name = table_name.capitalize()
+        metadata_table_name = table_name.capitalize() + "_metadata"
+        if conditions is None and meta_conditions is None:
+            return None
+
+        # we translate each condition into a single dict
+        #  conditions on columns
+        weaviate_conditions = []
+        if conditions is not None:
+            for condition in conditions:
+                column_key = condition.column
+                value_type = self._get_weaviate_value_type(condition.value)
+                weaviate_conditions.append(
+                    {
+                        "path": [column_key],
+                        "operator": self._get_weaviate_operator(condition.op),
+                        value_type: condition.value,
+                    }
+                )
+        # condition on metadata columns
+        if meta_conditions is not None:
+            for condition in meta_conditions:
+                meta_key = condition.column.split(".")[-1]
+                value_type = self._get_weaviate_value_type(condition.value)
+                weaviate_conditions.append(
+                    {
+                        "path": [
+                            "associatedMetadata",
+                            metadata_table_name,
+                            meta_key,
+                        ],
+                        "operator": self._get_weaviate_operator(condition.op),
+                        value_type: condition.value,
+                    }
+                )
+
+        # we combine all conditions into a single dict
+        all_conditions = (
+            {"operator": "And", "operands": weaviate_conditions}
+            if len(weaviate_conditions) > 1
```

Comment:
if weaviate_conditions

Reply:
resolved in b67cb1454fe8803bc304df7430cab6bcd5d670e7

Reply:
Here, if there are more than one conditions, then we are converting them into one single condition